src/dbSta/src/dbNetwork.cc

```diff
@@ -3109,60 +3109,46 @@ void dbNetwork::hierarchicalConnect(dbITerm* source_pin,
 // TODO: support finding uninstantiated modules
 dbModule* dbNetwork::findModule(const char* name)
 {
-  dbModule* module = nullptr;
-  Instance* top_inst = topInstance();
-  std::unique_ptr<InstanceChildIterator> child_iter{childIterator(top_inst)};
-  while (child_iter->hasNext()) {
-    Instance* child = child_iter->next();
-    if (network_->isHierarchical(child)) {
-      dbInst* db_inst;
-      dbModInst* mod_inst;
-      staToDb(child, db_inst, mod_inst);
-      if (mod_inst) {
-        dbModule* master = mod_inst->getMaster();
-        if (master) {
-          if (strcmp(master->getName(), name) == 0) {
-            module = master;
-            break;
+  // Helper function for recursive search
+  auto recursiveSearch
+      = [this](Instance* inst, const char* name, auto& self) -> dbModule* {
```

Comment:
Not essential but you could capture recursiveSearch rather than make it an argument, though you would have to declare its type rather than use auto.

Reply:
This API has been removed and replaced by dbBlock::findModule().